hoodie-cli/src/main/java/com/uber/hoodie/cli/commands/SparkMain.java

```diff
@@ -22,26 +22,52 @@
 import com.uber.hoodie.common.util.FSUtils;
 import com.uber.hoodie.config.HoodieIndexConfig;
 import com.uber.hoodie.config.HoodieWriteConfig;
+import com.uber.hoodie.configs.AbstractJobConfig;
+import com.uber.hoodie.configs.HDFSParquetImporterJobConfig;
+import com.uber.hoodie.configs.HoodieCommitRollbackJobConfig;
+import com.uber.hoodie.configs.HoodieCompactionAdminToolJobConfig;
+import com.uber.hoodie.configs.HoodieCompactorJobConfig;
+import com.uber.hoodie.configs.HoodieDeduplicatePartitionJobConfig;
+import com.uber.hoodie.configs.HoodieRollbackToSavePointJobConfig;
 import com.uber.hoodie.index.HoodieIndex;
-import com.uber.hoodie.io.compact.strategy.UnBoundedCompactionStrategy;
 import com.uber.hoodie.utilities.HDFSParquetImporter;
 import com.uber.hoodie.utilities.HoodieCompactionAdminTool;
-import com.uber.hoodie.utilities.HoodieCompactionAdminTool.Operation;
 import com.uber.hoodie.utilities.HoodieCompactor;
+import java.util.HashMap;
+import java.util.Map;
 import org.apache.log4j.Logger;
 import org.apache.spark.api.java.JavaSparkContext;
 import org.apache.spark.sql.SQLContext;
 
-public class SparkMain {
+public class SparkMain extends AbstractJobConfig {
 
-  protected static final Logger LOG = Logger.getLogger(SparkMain.class);
+  private static final Logger LOG = Logger.getLogger(SparkMain.class);
 
   /**
    * Commands
    */
   enum SparkCommand {
     ROLLBACK, DEDUPLICATE, ROLLBACK_TO_SAVEPOINT, SAVEPOINT, IMPORT, UPSERT, COMPACT_SCHEDULE, COMPACT_RUN,
-    COMPACT_UNSCHEDULE_PLAN, COMPACT_UNSCHEDULE_FILE, COMPACT_VALIDATE, COMPACT_REPAIR
+    COMPACT_UNSCHEDULE_PLAN, COMPACT_UNSCHEDULE_FILE, COMPACT_VALIDATE, COMPACT_REPAIR;
+  }
+
+  private Map<String, AbstractJobConfig> commandConfigMap = new HashMap<>();
```

Comment:
Can you make this a configFactoryMap ?

commandName -> configCreator (which is a function taking command-line  params and returning AbstractConfig.

Also, rename AbstractJobConfig to AbstractCommandConfig

Reply:
Made the changes as per the suggestion.